my-app/src/AddBox/AddBox.jsx

```diff
@@ -0,0 +1,180 @@
+import React, { Component } from 'react';
+import Autosuggest from 'react-autosuggest';
+
+//import isMobile from 'ismobilejs';
+
+//const focusInputOnSuggestionClick = !isMobile.any;
+const uniqLabelFast = a => {
+  /**
+   * Quickly makes a list tags without repeating labels
+   * @param a {[Tag]} list of tags
+   */
+
+    var seen = {};
+    var out = [];
+    var len = a.length;
+    var j = 0;
+    for(var i = 0; i < len; i++) {
+         var item = a[i].label;
+         if(seen[item] !== 1) {
+               seen[item] = 1;
+               out[j++] = a[i];
+         }
+    }
+    return out;
+}
+
+const escapeRegexCharacters = str => str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+
+const getSuggestionValue = (object) => {
+  if (object.hasOwnProperty('label')) {
+    return object.label;
+  } else if (object.hasOwnProperty('name')) {
+    return object.name;
+  }
+}
+
+const getSuggestions = (value, options) => {
+  const escapedValue = escapeRegexCharacters(value.trim());
+  if (escapedValue === '') {
+    return [];
+  }
+  const regex = new RegExp('^' + escapedValue, 'i');
+  return options.filter(tag_or_person => regex.test(getSuggestionValue(tag_or_person)));
+};
+
+
+const renderSuggestion = (suggestion, {query}) => {
+  var img = 'none';
+  if (suggestion.hasOwnProperty('label')) {
+    img = 'TAG';
+  } else if (suggestion.hasOwnProperty('name')) {
+    img = 'PERSON';
+  }
+  return(
+         //TODO: how do you get theme classes in?
+    <span className='sugestionContent'>{getSuggestionValue(suggestion) + ' ' + img}</span>
+    );
+}
+
+
+export default class AddBox extends Component {
+  constructor() {
+    super();
+    this.state = {
+      value: '',
+      suggestions: [],
+      tags:[],
+    }
+  }
+
+  componentWillMount = value => {
+    var options = uniqLabelFast(this.props.tags);
```

Comment:
I think that 'props' are passed in as an argument to the class constructor -- How about putting this in the initial value of state instead of waiting until the component is ready to be mounted into the render tree?

Reply:
Or just rely on componentWillReceiveProps alone.